### PR TITLE
Remove BC escaped double quote behavior

### PR DESCRIPTION
**Pull request by phinze**
When we switched to the current edition of the HCL parser, we
inadvertently broke some undocumented behavior where users could
successfully used escaped double quotes in a nested interpolation like
this:

    foo = "${somefunc(\"somearg\")}"

The proper syntax here is:

    foo = "${somefunc("somearg")}"

Because once you are inside the interpolation braces, the "quoting
context" is different.

At the time, while we didn't like the fact that the non-standard syntax
was in the wild, we didn't want to break BC, so we treated it as a bug
and fixed it in #62.

Now that we are at the moment of a major Terraform release, we can yank
the BC compatible behavior, which fixes a currently broken use case -
there is no way to express strings with double quotes in them from
inside interpolation braces.

    foo = "${somefunc("\"inquotes\"")}"

After merge and a dep upgrade, this will take care of
https://github.com/hashicorp/terraform/issues/5550

### CODE REVIEW

**Comment by phinze** [comment]
cc @jen20 or @jbardin for review

**Comment by mitchellh** [comment]
This seems reasonable to me. :+1:

File: hcl/strconv/quote.go [hunk 1]
```diff
@@ -70,7 +70,7 @@ func Unquote(s string) (t string, err error) {
 				if r == '\\' {
 					q, _ := utf8.DecodeRuneInString(s)
 					switch q {
-					case '"', '\\':
```
Comment (jbardin) [inline hcl/strconv/quote.go]: leftover comment above about leaving escaped double quotes

**Comment by phinze** [comment]
Forgot about @mitchellh :smile:

Good catch @jbardin! Repushed w/ fixed comments - PTAL

**Comment by jbardin** [comment]
LGTM

**Pull request merged**